src/gui/search/indexeroptionsdialog.h

```diff
@@ -31,30 +31,32 @@
 #include <QDialog>
 
 #include "base/settingvalue.h"
+#include "base/search/searchengine.h"
 
 namespace Ui
 {
-    class PluginSourceDialog;
+    class IndexerOptionsDialog;
 }
 
-class PluginSourceDialog final : public QDialog
+class IndexerOptionsDialog final : public QDialog
 {
     Q_OBJECT
-    Q_DISABLE_COPY_MOVE(PluginSourceDialog)
+    Q_DISABLE_COPY_MOVE(IndexerOptionsDialog)
 
 public:
-    explicit PluginSourceDialog(QWidget *parent = nullptr);
-    ~PluginSourceDialog() override;
+    explicit IndexerOptionsDialog(QWidget *parent = nullptr);
+    ~IndexerOptionsDialog() override;
 
-signals:
-    void askForUrl();
-    void askForLocalFile();
-
-private slots:
-    void on_localButton_clicked();
-    void on_urlButton_clicked();
+    QString indexerName() const;
+    void setIndexerName(const QString &name);
+    IndexerOptions indexerOptions() const;
+    void setIndexerOptions(const IndexerOptions &indexerOptions);
 
 private:
-    Ui::PluginSourceDialog *m_ui;
+    void loadState();
+    void saveState();
+    void validate();
```

Comment:
looks like a const function.

Reply:
>looks like a const function.

I still can't understand how you recognize those methods that should be `const`...
As for this. A function that doesn't return any result should at least have a side effect, otherwise it doesn't make sense, doesn't it?

Reply:
Sorry, false positive. It is unusual for me to see function of this name has side effects.

Reply:
>It is unusual for me to see function of this name has side effects.

Maybe you're right. But I can't think of a better name.

Reply:
>Maybe you're right. But I can't think of a better name.

To make it unconfusing, maybe emphasize the actions on the button? For example `setOKButtonEnabled`.